server/src/main/java/org/apache/iotdb/db/auth/ClusterAuthorityFetcher.java

```diff
@@ -0,0 +1,337 @@
+            AuthorOperator.AuthorType.values()[authorizerReq.getAuthorType()]
+                .toString()
+                .toLowerCase(Locale.ROOT),
+            authorizerResp.getStatus());
+        future.setException(new StatementExecutionException(authorizerResp.getStatus()));
+      } else {
+        future = AuthorizerManager.getInstance().buildTSBlock(authorizerResp.getAuthorizerInfo());
+      }
+    } catch (TException e) {
+      logger.error("Failed to connect to config node.");
+      authorizerResp.setStatus(
+          RpcUtils.getStatus(
+              TSStatusCode.EXECUTE_STATEMENT_ERROR, "Failed to connect to config node."));
+    }
+    return future;
+  }
+
+  @Override
+  public TSStatus checkUser(String username, String password) {
+    User user = userCache.getIfPresent(username);
+    if (user != null) {
+      if (password != null && AuthUtils.validatePassword(password, user.getPassword())) {
+        return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+      } else {
+        return RpcUtils.getStatus(
+            TSStatusCode.WRONG_LOGIN_PASSWORD_ERROR, "Authentication failed.");
+      }
+    } else {
+      TLoginReq req = new TLoginReq(username, password);
+      TPermissionInfoResp status = null;
+      try (ConfigNodeClient configNodeClient =
+          configNodeClientManager.borrowClient(ConfigNodeInfo.partitionRegionId)) {
+        // Send request to some API server
+        status = configNodeClient.login(req);
+      } catch (TException | IOException e) {
+        logger.error("Failed to connect to config node.");
+        status = new TPermissionInfoResp();
+        status.setStatus(
+            RpcUtils.getStatus(
+                TSStatusCode.EXECUTE_STATEMENT_ERROR, "Failed to connect to config node."));
+      } finally {
+        if (status == null) {
+          status = new TPermissionInfoResp();
+        }
+      }
+      if (status.getStatus().getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        userCache.put(username, cacheUser(status));
+        return status.getStatus();
+      } else {
+        return status.getStatus();
+      }
+    }
+  }
+
+  public TSStatus checkPath(String username, List<String> allPath, int permission) {
+    TCheckUserPrivilegesReq req = new TCheckUserPrivilegesReq(username, allPath, permission);
+    TPermissionInfoResp status = null;
+    try (ConfigNodeClient configNodeClient =
+        configNodeClientManager.borrowClient(ConfigNodeInfo.partitionRegionId)) {
+      // Send request to some API server
+      status = configNodeClient.checkUserPrivileges(req);
+    } catch (TException | IOException e) {
+      logger.error("Failed to connect to config node.");
+      status = new TPermissionInfoResp();
+      status.setStatus(
+          RpcUtils.getStatus(
+              TSStatusCode.EXECUTE_STATEMENT_ERROR, "Failed to connect to config node."));
+    } finally {
+      if (status == null) {
+        status = new TPermissionInfoResp();
+      }
```

Comment:
What is the meaning of this code？
And I think if this code is triggered, this method will return empty status contains nothing.

Reply:
Deleted~